actionners/kubernetes/drain/drain.go

```diff
@@ -0,0 +1,188 @@
+		if ignored {
+			// Append ignored pods info
+			aggregatedLog.Output += line.Output + "; "
+			ignoredPods = append(ignoredPods, p.Name)
+			continue
+		}
+		if err != nil {
+			allErrors = append(allErrors, err.Error())
+			continue
+		}
+
+		// Eviction logic if the pod is not ignored
+		err = performEviction(client, p, gracePeriodSeconds)
+		if err != nil {
+			allErrors = append(allErrors, err.Error())
+		}
+	}
+
+	if len(allErrors) > 0 {
+		aggregatedLog.Error = strings.Join(allErrors, "; ")
+		aggregatedLog.Status = "failure"
+	} else {
+		aggregatedLog.Status = "success"
+		aggregatedLog.Output = fmt.Sprintf("Node '%v' drained, ignored pods: %v", node.Name, strings.Join(ignoredPods, ", "))
+	}
+
+	return aggregatedLog, nil
+}
+
+func performEviction(client *kubernetes.Client, pod corev1.Pod, gracePeriodSeconds *int64) error {
+	delOpts := metav1.DeleteOptions{GracePeriodSeconds: gracePeriodSeconds}
+	evictionGroupVersion, err := CheckEvictionSupport(client)
+	if err != nil {
+		return err
+	}
+
+	switch evictionGroupVersion {
+	case policyv1.SchemeGroupVersion:
+		eviction := &policyv1.Eviction{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pod.Name,
+				Namespace: pod.Namespace,
+			},
+			DeleteOptions: &delOpts,
+		}
+		return client.PolicyV1().Evictions(eviction.Namespace).Evict(context.TODO(), eviction)
+	default:
+		eviction := &policyv1beta1.Eviction{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pod.Name,
+				Namespace: pod.Namespace,
+			},
+			DeleteOptions: &delOpts,
+		}
+		return client.Clientset.PolicyV1beta1().Evictions(pod.Namespace).Evict(context.Background(), eviction)
+	}
+}
+
+func CheckParameters(action *rules.Action) error {
+	parameters := action.GetParameters()
+
+	var config Config
+	err := utils.DecodeParams(parameters, &config)
+	if err != nil {
+		return err
+	}
+
+	err = utils.AddCustomValidation(validatorName, ValidateMinHealthyReplicas)
+	if err != nil {
+		return err
+	}
+
+	err = utils.ValidateStruct(config)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func ValidateMinHealthyReplicas(fl validator.FieldLevel) bool {
+	minHealthyReplicas := fl.Field().String()
+
+	reg := regexp.MustCompile(`\d+(%)?`)
+	result := reg.MatchString(minHealthyReplicas)
+	return result
+}
+
+func CheckEvictionSupport(client *kubernetes.Client) (schema.GroupVersion, error) {
+	discoveryClient := client.Clientset.Discovery()
+
+	// version info available in subresources since v1.8.0 in https://github.com/kubernetes/kubernetes/pull/49971
+	resourceList, err := discoveryClient.ServerResourcesForGroupVersion("v1")
+	if err != nil {
+		return schema.GroupVersion{}, err
+	}
+	for _, resource := range resourceList.APIResources {
+		if resource.Name == EvictionSubresource && resource.Kind == EvictionKind && len(resource.Group) > 0 && len(resource.Version) > 0 {
+			return schema.GroupVersion{Group: resource.Group, Version: resource.Version}, nil
+		}
+	}
+	return schema.GroupVersion{}, nil
+}
```

Comment:
I see your point, but the Eviction endpoint is v1 since K8s 1.22 which is 3y old. Can we consider to just use the v1 and avoid such complexity? Not an issue to specify a requirement of K8s > 1.25. The official docs of K8s only considers the last 5 versions (from 1.26 to 1.30 currently)